SimPEG/Maps.py

```diff
@@ -1120,6 +1285,99 @@ def deriv(self, m, v=None):
         return self.P
 
 
+class Mesh2MeshTopo(IdentityMap):
+    """
+        Takes a model on one mesh are translates it to another mesh
+        with consideration of topography
+
+    """
+    tree = None
+    nIterpPts = 6
+    P = None  #: The CSR projection matrix.
+    epsilon = 1e-8  # Small value to avoid 0 division in weights
+
+    def __init__(self, meshes, actinds, **kwargs):
+        Utils.setKwargs(self, **kwargs)
+
+        assert type(meshes) is list, "meshes must be a list of two meshes"
+        assert len(meshes) == 2, "meshes must be a list of two meshes"
+        assert type(actinds) is list, "actinds must be a list of two meshes"
+        assert len(actinds) == 2, "actinds must be a list of two meshes"
+        assert meshes[0].dim == meshes[1].dim, """The two meshes must be the same dimension"""
+
+        self.mesh = meshes[0]
+        self.mesh2 = meshes[1]
+        self.actind = actinds[0]
+        self.actind2 = actinds[1]
+        self._createProjection()
```

Comment:
ideally, we should not create the projection on instantiation, it should only be created when first asked for. This can be achieved through lazy loading
```
@property
def P(self):
    if get_attr(self, '_P', None) is None: 
        # code that creates Projection
        self._P = P
    return self.P
```